org.rdkit.knime.types/python3/knime/types/ext/rdkit.py

```diff
@@ -0,0 +1,214 @@
+
+@author Carsten Haubold, KNIME GmbH, Konstanz, Germany
+"""
+import logging
+import knime_types as kt
+
+LOGGER = logging.getLogger(__name__)
+
+try:
+    # We cannot import rdkit at the top of the file because it would then be required
+    # everywhere, even if we do not use the RDKit columns or don't even
+    # have an RDKit column in the table but the type is registered.
+    # All Python environments used in KNIME would then have to provide rdkit,
+    # which we don't want.
+    from rdkit import Chem
+    from rdkit.Chem.rdChemReactions import ChemicalReaction
+    from rdkit.DataStructs.cDataStructs import (
+        ExplicitBitVect,
+        UIntSparseIntVect,
+        IntSparseIntVect,
+        ULongSparseIntVect,
+        LongSparseIntVect,
+    )
+
+except ImportError as e:
+    LOGGER.info(
+        "RDKit type support not available because 'rdkit' could not be imported", e
+    )
+
+    # we define a dummy type Chem.Mol to be used instead.
+    # Any attempt to work with this data will fail.
+    class Chem:
+        class Mol:
+            pass
+
+    class ChemicalReaction:
+        pass
+
+    class ExplicitBitVect:
+        pass
+
+    class UIntSparseIntVect:
+        pass
+
+    class IntSparseIntVect:
+        pass
+
+    class LongSparseIntVect:
+        pass
+
+    class ULongSparseIntVect:
+        pass
+
+
+class RDKitMolAdapterValueFactory(kt.PythonValueFactory):
+    def __init__(self):
+        kt.PythonValueFactory.__init__(self, Chem.Mol)
+
+    def decode(self, storage):
+        if storage is None:
+            return None
+
+        value = Chem.Mol(storage["0"])
+        return value
+
+    def encode(self, value):
+        if value is None:
+            return None
+        return {"0": value.ToBinary(), "1": None}
+
+
+class RDKitMolValueFactory(kt.PythonValueFactory):
+    def __init__(self):
+        kt.PythonValueFactory.__init__(self, Chem.Mol)
+
+    def decode(self, storage):
+        if storage is None:
+            return None
+        return Chem.Mol(storage)
+
+    def encode(self, value):
+        if value is None:
+            return None
+        return value.ToBinary()
+
+
+class RDKitReactionValueFactory(kt.PythonValueFactory):
+    def __init__(self):
+        kt.PythonValueFactory.__init__(self, ChemicalReaction)
+
+    def decode(self, storage):
+        if storage is None:
+            return None
+        return ChemicalReaction(storage)
+
+    def encode(self, value):
+        if value is None:
+            return None
+        return value.ToBinary()
+
+
+class RDKitFingerprintValueFactory(kt.PythonValueFactory):
+    def __init__(self):
+        kt.PythonValueFactory.__init__(self, ExplicitBitVect)
+
+    def decode(self, storage):
+        if storage is None:
+            return None
+        import rdkit
+
+        length = int.from_bytes(storage[:8], byteorder="little")
+        fp = rdkit.DataStructs.CreateFromBinaryText(storage[8:])
```

Comment:
Why not import `CreateFromBinaryText` when the other RDKit bits are imported towards the top of the file?
If you do this, you can also drop the `import rdkit` on line 156.